tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/VoteAPIHelper.java

```diff
@@ -137,6 +137,13 @@ private static Response handleGetRows(ArgsForGet args) {
                 if (xp == null) {
                     return Response.status(404).entity(new STError(ErrorCode.E_BAD_XPATH)).build();
                 }
+                // TODO: only return data for the single row the front end requested!
+                // This matcher can match "alt" paths related to the given path, so for example,
+                // if the requested path is
+                // //ldml/localeDisplayNames/territories/territory[@type="FK"]
+                // then currently we return not only data for that path, but also
+                // for //ldml/localeDisplayNames/territories/territory[@type="FK"][@alt="variant"]
+                // Reference: https://unicode-org.atlassian.net/browse/CLDR-16766
                 matcher = XPathMatcher.getMatcherForString(xp); // single string
```

Comment:
I see, it's not an exact match.  Maybe need an exactMatcherForString(). or just add `$` to the end of the xpath?

Reply:
exactMatcherForString sounds good -- might be more efficient than a regex ending with $

this is the only place getMatcherForString is called

BaseAndPrefixMatcher is deprecated, and seems not to be used anywhere other than getMatcherForString so it looks like we could replace getMatcherForString with a new exactMatcherForString and remove BaseAndPrefixMatcher

Reply:
> BTW any idea what DataPage.CONTINENT_DIVIDER "~" is about? I put a breakpoint here after the "if" and tried voting in all the timzezone paths but I never hit it
> 
> ```
>    public static XPathMatcher getMatcherForString(String str) {
>         if (str.contains(DataPage.CONTINENT_DIVIDER)) {
>             // just return prefix
>             return BaseAndPrefixMatcher.getInstance(
>                     XPathTable.NO_XPATH, str.substring(0, str.indexOf(DataPage.CONTINENT_DIVIDER)));
>         } else {
>             return BaseAndPrefixMatcher.getInstance(XPathTable.NO_XPATH, str);
>         }
>     }
> ```

CONTENT_DIVIDER… isnt' that the rocky mountains?

Reply:
OK seriously. getMatcherForString()  was added in 2012 may have had other callers. 

I would suspect that it may have been used before there was a 'fetch by pathheader section' api.  But don't remember exactly.   @macchiati ?

Reply:
... where the rivers change directions ...